src/libraries/Microsoft.PowerFx.Core/Public/Types/IAggregateTypeIdentity.cs

```diff
@@ -0,0 +1,18 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+namespace Microsoft.PowerFx.Types
+{
+    [System.Diagnostics.CodeAnalysis.SuppressMessage(
+        "Design",
+        "CA1040:Avoid empty interfaces", 
+        Justification = "Intentional. Used as a marker for identity of user implemented aggregate types")]
+    
+    // $$$ This isn't enforced/enforcable, thoughts?
+    [ThreadSafeImmutable] 
+    public interface IAggregateTypeIdentity
+    {
+        // $$$ Maybe this should be an abstract class, since we could force implementers to override Equals/GetHashCode?
```

Comment:
>Equals/GetHashCode? [](http://example.com/codeflow?start=101&length=19)

Can we do that on an interface?

The main purpose is equality, so those would make it explicit. 

Then we could also add an explicit ToString  that we may find useful in error messages, diagnostics.

Reply:
I'll add Equals/GetHashCode/ToString to the interface and make it clear that it's expected, but it won't actually force implementers to override them since every object already implements them.